ConnectN/alpha_beta_agent.py

```diff
@@ -29,9 +29,88 @@ def __init__(self, name, max_depth):
     def go(self, brd):
         """Search for the best move (choice of column for the token)"""
         # Your code here
-        parent_node = alpha_beta_node.AlphaBetaNode(brd, None, None)
-        parent_node.children = self.make_children(parent_node, 0)
-        return self.find_max_node(parent_node).col
+        move_col = self.alpha_beta_search(brd)
+        return move_col
+
+    # The first level of an Alpha-Beta search that kicks off the rest.
+    # Separated because the moves need to be tracked at the first level
+    #
+    # PARAM [board.Board] brd: the board state
+    # RETURN [int]: the column of the move evaluated to be best
+    def alpha_beta_search(self, brd):
+        # Array of possible moves
+        possible_moves = []
+        successors = self.get_successors(brd)
+        # For each possible move
+        for successor in successors:
+            brd = successor[0]
+            col = successor[1]
+            # create new node, evaluation created by alpha-beta searching all possible children to max_depth
+            new_node = alpha_beta_node.AlphaBetaNode(brd, col, self.min_value(brd, self.max_depth, 0, 1))
+            # check if the move wins and if so return immediately
+            if new_node.evaluation == 1:
+                return new_node.col
+            # Add to the array
+            possible_moves.append(new_node)
+        # Assign Starting Vals to find the best move
+        highest_move_val = 0
+        move_col = 1
+        # Find the column associated with the best move
+        for move in possible_moves:
```

Comment:
I think instead of having this for loop we can instead have a tuple before we go through the successors that has max eval score and the column associated with it and it updates after every successor.